modules/apps/data-engine/data-engine-rest-test/src/testIntegration/java/com/liferay/data/engine/rest/resource/v2_0/test/util/DataDefinitionTestUtil.java

```diff
@@ -79,6 +81,40 @@ public static DataDefinition addDataDefinitionWithDataLayout(long groupId)
 			groupId, "app-builder", dataDefinition);
 	}
 
+	public static DataDefinition addDataDefinitionWithFieldSet(long groupId)
+		throws Exception {
+
+		DataDefinitionResource.Builder builder =
+			DataDefinitionResource.builder();
+
+		DataDefinitionResource dataDefinitionResource = builder.authentication(
+			"test@liferay.com", "test"
+		).locale(
+			LocaleUtil.getDefault()
+		).build();
+
+		DataDefinition fieldsetDataDefinition =
+			dataDefinitionResource.postSiteDataDefinitionByContentType(
+				groupId, "app-builder-fieldset",
+				DataDefinition.toDTO(read("data-definition-basic.json")));
+
+		DataDefinition dataDefinition = DataDefinition.toDTO(
```

Comment:
I'm guessing you discussed this with @gabrielwas but i was wondering i we dont have a single (new) json with fieldet instead of composing the object with two jsons

Reply:
Hum... It's because we need to save a fieldset before saving a data definition that uses it. So, I can't visualize a way to do that with a single json.

Reply:
Huumm. Got it :) No need then ^^